src/control/vendor/github.com/daos-stack/go-spdk/spdk/src/nvme_control.c

```diff
@@ -193,22 +174,28 @@ collect(struct ret_t *ret)
 			return;
 		}
 
-		cdata = spdk_nvme_ctrlr_get_data(ns_entry->ctrlr);
-
 		ns_tmp->id = spdk_nvme_ns_get_id(ns_entry->ns);
 		// capacity in GBytes
-		ns_tmp->size = spdk_nvme_ns_get_size(ns_entry->ns) / \
+		ns_tmp->size = spdk_nvme_ns_get_size(ns_entry->ns) /
 			       NVMECONTROL_GBYTE_BYTES;
 
 		pci_dev = spdk_nvme_ctrlr_get_pci_device(ns_entry->ctrlr);
 		if (!pci_dev) {
-			snprintf(ret->err, sizeof(ret->err), "get_pci_device");
+			snprintf(ret->err, sizeof(ret->err),
+				 "%s: get_pci_device", __func__);
+
 			ret->rc = -NVMEC_ERR_GET_PCI_DEV;
 			return;
 		}
 
-		if (set_pci_addr(pci_dev, ns_tmp->ctrlr_pci_addr,
-				 sizeof(ns_tmp->ctrlr_pci_addr), ret) != 0) {
+		pci_addr = spdk_pci_device_get_addr(pci_dev);
+		rc = spdk_pci_addr_fmt(ns_tmp->ctrlr_pci_addr,
+				       sizeof(ns_tmp->ctrlr_pci_addr),
+				       &pci_addr);
+		if (rc != 0) {
+			snprintf(ret->err, sizeof(ret->err),
+				 "spdk_pci_addr_fmt: rc %d", rc);
+			ret->rc = -NVMEC_ERR_PCI_ADDR_FMT;
```

Comment:
I'm not sure if I fully understand this, why do we need to duplicate the pci address of controller into each ns?

Reply:
for lookup purposes, this is how I'm associating each ns with the parent controller